src/Session/SessionManager.cc

```diff
@@ -531,7 +531,8 @@ void SessionManager::OnMessage(WSType* ws, std::string_view sv_message, uWS::OpC
             auto t_session = session->GetLastMessageTimestamp();
             auto t_now = std::chrono::high_resolution_clock::now();
             auto dt = std::chrono::duration_cast<std::chrono::seconds>(t_now - t_session);
-            if ((_settings.idle_session_wait_time > 0) && (dt.count() >= _settings.idle_session_wait_time)) {
+            if ((ProgramSettings::GetInstance().idle_session_wait_time > 0) &&
+                (dt.count() >= ProgramSettings::GetInstance().idle_session_wait_time)) {
```

Comment:
I suggest adding a `_settings` property to this class as well. Since it currently only uses one setting, we could also just add that as a property directly, but this is more consistent and future-proof (we may use other settings in the future).